vendor/bat-native-ledger/src/bat/ledger/internal/request/post_connect/uphold/post_connect_uphold.cc

```diff
@@ -0,0 +1,91 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ledger/internal/request/post_connect/uphold/post_connect_uphold.h"
+
+#include <map>
+#include <utility>
+
+#include "base/base64.h"
+#include "base/json/json_writer.h"
+#include "bat/ledger/internal/common/security_util.h"
+#include "bat/ledger/internal/ledger_impl.h"
+
+namespace ledger::request::connect {
+
+PostConnectUphold::PostConnectUphold(LedgerImpl* ledger, std::string&& address)
+    : PostConnect(ledger), address_(std::move(address)) {}
+
+PostConnectUphold::~PostConnectUphold() = default;
+
+base::expected<std::string, base::Location> PostConnectUphold::Content() const {
+  const auto wallet = ledger_->wallet()->GetWallet();
+  if (!wallet) {
+    BLOG(0, "Rewards wallet is null!");
+    return base::unexpected(base::Location::Current());
+  }
+
+  base::Value::Dict denomination;
+  denomination.Set("amount", "0");
+  denomination.Set("currency", "BAT");
+
+  base::Value::Dict body;
+  body.Set("denomination", std::move(denomination));
+  body.Set("destination", address_);
+
+  std::string octets;
+  if (!base::JSONWriter::Write(body, &octets)) {
+    BLOG(0, "Failed to write octets to JSON!");
+    return base::unexpected(base::Location::Current());
+  }
+
+  std::string digest = util::Security::DigestValue(octets);
+  std::string signature = util::Security::Sign(
+      {{{"digest", digest}}}, "primary", wallet->recovery_seed);
+  if (signature.empty()) {
+    BLOG(0, "Failed to create signature!");
+    return base::unexpected(base::Location::Current());
+  }
+
+  base::Value::Dict headers;
+  headers.Set("digest", std::move(digest));
+  headers.Set("signature", std::move(signature));
+
+  base::Value::Dict request;
+  request.Set("body", std::move(body));
+  request.Set("headers", std::move(headers));
+  request.Set("octets", std::move(octets));
+
+  std::string json_request;
+  if (!base::JSONWriter::Write(std::move(request), &json_request)) {
+    BLOG(0, "Failed to write request to JSON!");
+    return base::unexpected(base::Location::Current());
+  }
+
+  std::string signedLinkingRequest;
+  base::Base64Encode(std::move(json_request), &signedLinkingRequest);
+
+  base::Value::Dict content;
+  content.Set("signedLinkingRequest", std::move(signedLinkingRequest));
+
+  std::string json;
+  if (!base::JSONWriter::Write(std::move(content), &json)) {
+    BLOG(0, "Failed to write content to JSON!");
+    return base::unexpected(base::Location::Current());
+  }
+
+  return json;
+}
+
+base::expected<std::vector<std::string>, base::Location>
+PostConnectUphold::Headers() const {
+  return RequestBuilder::Headers();
```

Comment:
Is this idea here that the uphold version does not need headers, and should therefore bypass the implementation in `PostConnect`? Instead of skipping an override level, would it be clearer to just return an empty vector here?

Reply:
Yeah, sure.